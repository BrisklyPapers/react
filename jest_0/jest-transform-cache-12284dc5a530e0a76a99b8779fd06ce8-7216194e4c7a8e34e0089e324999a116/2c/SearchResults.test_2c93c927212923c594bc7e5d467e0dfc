'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/containers/SearchResults');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _enzyme = require('enzyme');
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/SearchResults', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ searchResults: { documents: [{ fileName: "foo.pdpf" }] } });
        store.dispatch = jest.fn();
        wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_SearchResults2.default, { store: store }));
    });

    it('maps state and dispatch to props', function () {
        expect(wrapper.props()).toEqual(expect.objectContaining({
            documents: [{ fileName: "foo.pdpf" }] }));

    });
});