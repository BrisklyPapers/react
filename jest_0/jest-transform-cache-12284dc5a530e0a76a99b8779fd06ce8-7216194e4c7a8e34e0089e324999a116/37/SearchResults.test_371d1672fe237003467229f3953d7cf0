'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/components/SearchResults.js');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/SearchResults', function () {
    it('renders correctly', function () {
        var searchResults = _reactTestRenderer2.default.create(
        _react2.default.createElement(_SearchResults2.default, { documents: [] }));

        expect(searchResults.toJSON()).toMatchSnapshot();
    });

    it('contains some search results', function () {
        var documents = [
        {
            fileName: 'bill.pdf',
            url: 'http://briskly-papers.com',
            text: 'testing',
            tags: ['bill', 'source'] }];


        var searchResults = _reactTestRenderer2.default.create(
        _react2.default.createElement(_SearchResults2.default, { documents: documents }));

        expect(searchResults.toJSON()).toMatchSnapshot();
    });
});