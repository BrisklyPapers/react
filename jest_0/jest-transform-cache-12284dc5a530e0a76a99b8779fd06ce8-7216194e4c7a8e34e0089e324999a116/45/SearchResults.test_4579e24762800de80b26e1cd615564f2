'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/containers/SearchResults');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('containers/SearchResults', function () {
    it('should render correctly', function () {
        var wrapper = _reactTestRenderer2.default.create(_react2.default.createElement(_SearchResults2.default, null));
        expect(_reactTestRenderer2.default.toJSON()).toMatchSnapShot();
    });
});