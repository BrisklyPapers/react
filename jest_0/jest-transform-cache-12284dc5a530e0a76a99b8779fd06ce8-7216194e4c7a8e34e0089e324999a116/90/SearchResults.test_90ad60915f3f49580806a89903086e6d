'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/containers/SearchResults');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/SearchResults', function () {
    it('should render correctly', function () {
        var store = mockStore({ searchResults: { documents: [] } });
        var wrapper = _reactTestRenderer2.default.create(_react2.default.createElement(_SearchResults2.default, { store: store }));
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should add documents to SearchResults', function () {
        var documents = [
        {
            fileName: 'bill.pdf',
            url: 'http://briskly-papers.com',
            text: ['testing search results'],
            tags: ['bill', 'source'] },

        {
            fileName: 'flower.jpg',
            url: 'https://localhost/search?q=flowers',
            text: ['pictures showing some flowers'],
            tags: ['flower', 'summer'] }];


        var store = mockStore({ searchResults: { documents: documents } });
        var wrapper = _reactTestRenderer2.default.create(_react2.default.createElement(_SearchResults2.default, { store: store }));
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});