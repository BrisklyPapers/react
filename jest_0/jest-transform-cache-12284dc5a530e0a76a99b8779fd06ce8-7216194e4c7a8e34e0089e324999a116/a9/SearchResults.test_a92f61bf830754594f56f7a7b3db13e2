'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/containers/SearchResults');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('containers/SearchResults', function () {
    var wrapper = void 0,store = void 0;

    beforeEach(function () {
        store = mockStore({ searchResults: { documents: [] } });
        store.dispatch = jest.fn();
        wrapper = shallow(_react2.default.createElement(_SearchResults2.default, { store: store }));
    });

    it('maps state ');
});