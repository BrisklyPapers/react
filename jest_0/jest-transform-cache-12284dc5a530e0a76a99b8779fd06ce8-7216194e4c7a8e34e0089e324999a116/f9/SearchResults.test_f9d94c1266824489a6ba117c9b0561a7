'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _SearchResults = require('../../src/containers/SearchResults');var _SearchResults2 = _interopRequireDefault(_SearchResults);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('containers/SearchResults', function () {
   it('should render correctly', function () {

   });
});