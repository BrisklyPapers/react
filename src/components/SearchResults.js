import React from 'react';
import SearchResult from './SearchResult';
import PropTypes from 'prop-types';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

const SearchResults = ({documents}) => {
    return (
        <MuiThemeProvider>
            <div>
                {documents.map(document => (
                    <SearchResult key={document.url} title={document.fileName} url={document.url} description={document.text.join()} />
                ))}
            </div>
        </MuiThemeProvider>
    );
};

SearchResults.propTypes = {
    documents: PropTypes.arrayOf(
        PropTypes.shape({
            fileName: PropTypes.string.isRequired,
            url: PropTypes.string.isRequired,
            text: PropTypes.arrayOf(
                PropTypes.string
            ).isRequired,
            tags: PropTypes.arrayOf(
                PropTypes.string
            ).isRequired
        }).isRequired
    ).isRequired
};

export default SearchResults;