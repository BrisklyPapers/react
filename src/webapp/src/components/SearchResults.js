import React from 'react'
import SearchResult from './SearchResult';

const SearchResults = () => {
    return (
        <div>
            <SearchResult title="test" url="http://test.de" description="Ich bin <em>ein Test</em>"/>
        </div>
    )
};

export default SearchResults