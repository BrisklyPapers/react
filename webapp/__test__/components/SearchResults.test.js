import React from 'react';
import SearchResults from '../../src/components/SearchResults.js';
import renderer from 'react-test-renderer';

describe('components/SearchResults', () => {
    it('renders correctly', () => {
        const documents = [];
        const searchResults = renderer.create(
            <SearchResults documents={[]}/>
        );
        expect(searchResults.toJSON()).toMatchSnapshot();
    });
});