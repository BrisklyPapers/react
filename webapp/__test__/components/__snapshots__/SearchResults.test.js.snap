// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchResults renders correctly 1`] = `<div />`;
